packages/next/src/client/components/react-dev-overlay/internal/helpers/group-stack-frames-by-module.ts

```diff
@@ -1,29 +1,17 @@
 import type { OriginalStackFrame } from './stack-frame'
 
 export type StackFramesGroup = {
-  framework?: 'next' | 'react'
+  moduleGroup?: OriginalStackFrame['sourceModule']
   stackFrames: OriginalStackFrame[]
 }
 
-/**
- * Get the origin framework of the stack frame by package name.
- */
-function getFramework(
-  sourcePackage: string | undefined
-): StackFramesGroup['framework'] {
-  if (!sourcePackage) return undefined
-
-  if (
-    /^(react|react-dom|react-is|react-refresh|react-server-dom-webpack|react-server-dom-turbopack|scheduler)$/.test(
-      sourcePackage
-    )
-  ) {
-    return 'react'
-  } else if (sourcePackage === 'next') {
-    return 'next'
-  }
-
-  return undefined
+/** TODO: Move this to react-dev-overlay/server/middleware.ts */
+function getModuleGroup(file: string | null): StackFramesGroup['moduleGroup'] {
+  if (!file) return
+  // Should we maybe want to group Node.js errors too?
+  // if (file.startsWith('node:internal/')) return 'node'
```

Comment:
In my opinion, instead of hiding Node.js internal errors (as done [here](https://github.com/vercel/next.js/pull/62206/files#diff-e7e732f178d05327d5baa8c14df25e8eb650ca63d78ba31f9c3635b76c51606dR52)), we could rather group these similar to Next.js and React errors, so the user still can dig in, if they want to.

Another possibility is a toggle/switch to show the "raw" error stack.

This needs a discussion first and can be done in a follow-up PR instead.